### ISSUE TITLE
Convergence problem about the joint supervision of CosFace and Softmax loss 

### ISSUE THREAD

**Issue by HanGuangXin** [layumi/Person_reID_baseline_pytorch#304]
Hi, there! Thanks for your amazing work!

But I have a question about CosFace loss.
Your code use joint supervision of of CosFace and Softmax loss. So I train the model like 'different losses' chapter, using the following command:
python train.py --warm_epoch 5 --stride 1 --erasing_p 0.5 --batchsize 32 --lr 0.08 --name warm5_s1_b32_lr8_p0.5_cosface100 --cosface --total 100; 

Then I get the following result:
Rank@1:0.904097 Rank@5:0.965261 Rank@10:0.977435 mAP:0.764123

But when I only use CosFace loss, and keep other settings unchanged, I am faced with a convergence problem, e.g. the train and val accuracy are very low, like 0.0013.

if opt.cosface:
    loss = criterion_cosface(ff, labels)/now_batch_size         # without CE loss

ArcFace paper says CosFace loss and ArcFace loss do not need Softmax.
So can you please explain why there is a convergence problem when only using CosFace as loss function?

**Comment by HanGuangXin** [comment]
And arcface loss has the same issue...

**Comment by layumi** [comment]
Hi @HanGuangXin ,
I do not change the gamma (also named as scale) in the two losses, i.e., cos face and arc face. 
The hyper-parameters are needed to be further tuned.

**Comment by HanGuangXin** [comment]
Thanks for your reply! I'll look into it if I had enough time.
